src/lib/actions/RuleListAction.ts

```diff
@@ -0,0 +1,73 @@
+import {Action} from "../ScannerCommand";
+import {Inputs, Rule} from "../../types";
+import {AnyJson} from "@salesforce/ts-types";
+import {Ux} from "@salesforce/sf-plugins-core";
+import {BundleName, getMessage} from "../../MessageCatalog";
+import {RuleFilter} from "../RuleFilter";
+import {Controller} from "../../Controller";
+import {Display} from "../Display";
+import {RuleFilterFactory} from "../RuleFilterFactory";
+
+/**
+ * The Action behind the "rule list" command
+ */
+export class RuleListAction implements Action {
+	private readonly display: Display;
+	private readonly ruleFilterFactory: RuleFilterFactory;
+
+	public constructor(display: Display, ruleFilterFactory: RuleFilterFactory) {
+		this.display = display;
+		this.ruleFilterFactory = ruleFilterFactory;
+	}
+
+	public validateInputs(_inputs: Inputs): Promise<void> { // eslint-disable-line @typescript-eslint/no-unused-vars
+		// Currently there is nothing to validate
+		return Promise.resolve();
+	}
+
+	public async run(inputs: Inputs): Promise<AnyJson> {
+		const ruleFilters: RuleFilter[] = this.ruleFilterFactory.createRuleFilters(inputs);
+
+		// TODO: Inject RuleManager as a dependency to improve testability by removing coupling to runtime implementation
+		const ruleManager = await Controller.createRuleManager();
+
+		const rules = await ruleManager.getRulesMatchingCriteria(ruleFilters);
+		this.display.displayTable(rules, columns);
+		// If the --json flag was used, we need to return a JSON. Since we don't have to worry about displayability, we can
+		// just return the filtered list instead of the formatted list.
+		return rules;
+	}
+}
+
+const MSG_YES: string = getMessage(BundleName.List, 'yes');
```

Comment:
My personal preference would be for file-scoped constants to be declared above the classes, but that's just me.

Reply:
That's fine with me. I'll make the update.

Reply:
Fixed - squashed into part 10 with rebase and then force push.